src/lib.rs

```diff
@@ -69,15 +69,96 @@
     unused_parens,
     while_true
 )]
-use ansi_term::{
-    Color::{Cyan, Purple, Red},
-    Style,
-};
 use std::env;
 use std::fmt;
 use std::fs::File;
 use std::io::{BufRead, BufReader, ErrorKind};
 use tracing_error::SpanTrace;
+use once_cell::sync::OnceCell;
+use owo_colors::{style, Style, XtermColors};
+
+static STYLES: OnceCell<Styles> = OnceCell::new();
+
+// XXX now that we support text effects, "color scheme" doesn't seem accurate anymore. Let me know if you'd prefer a different name. The good thing about `Styles` is, that it's pretty short.
+
+/// A struct that represents styles that should be used by `color_spantrace`
+#[derive(Debug, Copy, Clone, Default)]
+pub struct Styles {
+    file: Style,
+    line_number: Style,
+    target: Style,
+    fields: Style,
+    active_line: Style,
+}
+
+impl Styles {
+    /// Create blank styles
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Styles for a dark background. This is the default
+    pub fn dark() -> Self {
+        Self {
+            file:         style().purple(),
+            line_number:  style().purple(),
+            active_line:  style().white().bold(),
+
+            // XXX your trait from this file (which I have removed now) produced "xterm" colors. `owo_colors` documents the `xterm` module as follows: "XTerm 256-bit colors. Not as widely supported as standard ANSI but contains 240 more colors." (see *). Maybe it would be best to switch to something else now. In this case, the colors also have to be changed in `owo_colors`
+            // * https://github.com/jam1garner/owo-colors/blob/c0db9127cbc355dec36fbfa45b9adc534b72c4ad/src/colors.rs#L173
+            target:       style().color(XtermColors::UserBrightRed),
+            fields:       style().color(XtermColors::UserBrightCyan),
+        }
+    }
+
+    // XXX same as with `light` in `color_eyre`
+    /// Styles for a light background
+    pub fn light() -> Self {
+        Self {
+            file:         style().purple(),
+            line_number:  style().purple(),
+            target:       style().red(),
+            fields:       style().blue(),
+            active_line:  style().bold(),
+        }
+    }
+
+    /// Styles printed paths
+    pub fn file(mut self, style: Style) -> Self {
+        self.file = style;
+        self
+    }
+
+    /// Styles the line number of a file
+    pub fn line_number(mut self, style: Style) -> Self {
+        self.line_number = style;
+        self
+    }
+
+    /// Styles the target (i.e. the module and function name, and so on)
+    pub fn target(mut self, style: Style) -> Self {
+        self.target = style;
+        self
+    }
+
+    // XXX is this correct?
+    /// Styles fields of the `tracing` crate (in the context of `color_spantrace`, the arguments of functions and methods)
```

Comment:
Its not quite right, spans can be created independently of functions / methods by just invoking the `span!` macro directly 

```suggestion
    /// Styles fields associated with a the `tracing::Span`.
```

Reply:
👍